models/contacts.js

```diff
@@ -1,19 +1,67 @@
 const fs = require('fs/promises')
+const path = require('path')
+const { nanoid } = require('nanoid')
 
-const listContacts = async () => {}
+const contactsPath = path.join(__dirname, 'contacts.json')
 
-const getContactById = async (contactId) => {}
+const updateContact = async (contacts) => {
+  await fs.writeFile(contactsPath, JSON.stringify(contacts, null, 2));
+}
 
-const removeContact = async (contactId) => {}
+const listContacts = async () => {
+  const data = await fs.readFile(contactsPath);
+  return JSON.parse(data);
+}
 
-const addContact = async (body) => {}
+const getContactById = async (id) => {
+  const contacts = await listContacts();
+  const result = contacts.find((item) => item.id === `${id}`);
+  if (!result) {
+    return null;
+  }
+  return result;
+}
 
-const updateContact = async (contactId, body) => {}
+const removeContact = async (id) => {
+  const contacts = await listContacts();
+  const idx = contacts.findIndex((item) => item.id === `${id}`);
+  if (idx === -1) {
+    return null;
+  }
+  const [removeContact] = contacts.splice(idx, 1);
```

Comment:
Nie lepiej cały mechanizm usuwania zrobiź za pomocą filter?

Reply:
Jak działa to niech już zostanie jak jest ;)